### ISSUE TITLE
Black Holes break map flow and suck in actors that are away from the player

### ISSUE THREAD

**Issue by JMartinez9820** [pa1nki113r/Project_Brutality#1204]
Black Holes suck in any actor, even if they're very far away from the player, such as in a teleporter. I would suggest having a check that can only suck in any actor that is visible to the player. Or just decrease the force magnitude of the Black hole attack.

**Comment by inkoalawetrust** [comment]
Do you mean that it sucks in teleporter spots ? I think I already published a fix for that. The problem was that the teleport spots were in the blockmap for some reason. And sucking in only actors visible to the player sounds like a really bad idea that would lead to blatantly and very visibly wrong behavior, like a blackhole not sucking monsters behind a wall until they get in sight of you.

**Comment by JMartinez9820** [comment]
No. What I meant is that in some occasions, the black hole pull force is so strong that even monsters in different rooms get pulled in. Here is a comparison:

This room from a teleporter in Doom 2 Map 23 before the primary black hole:
![Screenshot_Doom_20231129_215214](https://github.com/pa1nki113r/Project_Brutality/assets/40510256/62fd6dad-cbed-4862-b2c9-030c496135f1)

The same room, but after the black hole shot if fired CLOSE to the first teleporter that leads to said room:
![Screenshot_Doom_20231129_215207](https://github.com/pa1nki113r/Project_Brutality/assets/40510256/0a1f6d1c-b345-4392-a1c4-e1c69850bed8)

The hell knights and chaingunners get pulled to the left side along with same barrels being destroyed in said room. Although the new black hole is likely a more realistic approach, this can be very game breaking in some elements.
